src/XMonad/Core.hs

```diff
@@ -338,6 +338,9 @@ class (Show (layout a), Typeable layout) => LayoutClass layout a where
     description :: layout a -> String
     description      = show
 
+    {-# MINIMAL (runLayout | ((doLayout | pureLayout) , emptyLayout))
```

Comment:
Did you test this with xmonad-contrib? Few layouts override emptyLayout, so this will result in lots of warnings.

It seems to me the minimal complete definition of a LayoutClass is the one that overrides nothing, as `instance LayoutClass Full` shows, and that it is completely fine that way.

Reply:
Mh you're right.  I guess we should change the comment then?  Something along the lines of highlighting the dependencies of the different functions?

Reply:
Okay I pushed a doc change instead